### PR TITLE
Prevent to modify readonly fields (fixes #26)

### LINKED ISSUE #26: Make sure some attributes cannot be changed by the client.
Some attributes are immutable:

* ``url``
* ``stored_on``
* ~~``resolved_url``~~
* ~~``resolved_title``~~
* ``last_modified``

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

File: readinglist/resource.py [hunk 1]
```diff
@@ -93,6 +95,12 @@ class RessourceSchema(colander.MappingSchema):
     _id = colander.SchemaNode(colander.String(), missing=colander.drop)
     last_modified = TimeStamp()
 
+    __readonly_fields__ = ('_id', 'last_modified')
```
Comment (tarekziade) [inline readinglist/resource.py]: why do you use double-dunder ? is it something from Colander ? if not it's potential trouble when accessing it

Comment (Natim) [inline readinglist/resource.py]: Yes I had the same strange feeling reading this.

Comment (leplatrem) [inline readinglist/resource.py]: Just a naive inspiration to distinguish this from actual schema fields :/

Any alternative idea ?


```python
class Schema(...):
    field1 = ...
    field2 = ...

    class Options:
        readonly_fields = ...
```

Comment (Natim) [inline readinglist/resource.py]:
```
class Schema(...):
    field1 = ...
    field2 = ...

    class Meta:
        readonly_fields = ...
```

Looks like Django ideas are coming.

Maybe just `_readonly_fields`?

File: readinglist/errors.py [hunk 2]
```diff
@@ -92,3 +92,11 @@ def json_error(errors):
     response.status = errors.status
 
     return response
+
+
+class ImmutableFieldError(Exception):
+    """Raised when attempting to modify a read-only field."""
+    def __init__(self, field, msg=None):
+            self.field = field
```
Comment (ametaireau) [inline readinglist/errors.py]: It seems we have too many spaces here.

File: readinglist/resource.py (hunk 1 above)
Comment (ametaireau) [inline readinglist/resource.py]: I actually quite like the `__readonly_fields__`, I find its meaning obvious, and says to me "this is a metadata field". I don't know where this is coming from though :D

Comment (ametaireau) [inline readinglist/resource.py]: Having an `Options` or `Meta` subclass seems also good to me, and it seems easy to have interchangeable subclasses (with different readonly_fields for instance, or which computes the fields only when accessed).

**Pull request merged**